src/main/java/org/reactome/addlinks/referencecreators/SimpleReferenceCreator.java

```diff
@@ -130,19 +126,25 @@ public void createIdentifiers(long personID, Map<String, T> mapping, List<GKInst
 						}
 					}
 
-					String targetRefDBIdentifier = (String)mapping.get(sourceReferenceIdentifier);
-					this.logger.trace("{} ID: {}; {} ID: {}", this.sourceRefDB, sourceReferenceIdentifier, this.targetRefDB, targetRefDBIdentifier);
-					// Look for cross-references.
-					boolean xrefAlreadyExists = checkXRefExists(sourceReference, targetRefDBIdentifier);
-					if (!xrefAlreadyExists)
-					{
-						this.logger.trace("\tCross-reference {} does not yet exist, need to create a new identifier!", targetRefDBIdentifier);
-						sourceIdentifiersWithNewIdentifier.incrementAndGet();
-						thingsToCreate.add(targetRefDBIdentifier+","+String.valueOf(sourceReference.getDBID())+","+speciesID);
+					// The T value of the mapping can be either a List or a String depending on the FileProcessors invoked.
+					Set<String> targetRefDBIdentifiers = new HashSet<>();
+					if (mapping.get(sourceReferenceIdentifier) instanceof List) {
+						targetRefDBIdentifiers.addAll((List) mapping.get(sourceReferenceIdentifier));
+					} else {
+						targetRefDBIdentifiers.add((String) mapping.get(sourceReferenceIdentifier));
 					}
-					else
-					{
-						sourceIdentifiersWithExistingIdentifier.incrementAndGet();
+
+					for (String targetRefDBIdentifier : targetRefDBIdentifiers) {
+						this.logger.trace("{} ID: {}; {} ID: {}", this.sourceRefDB, sourceReferenceIdentifier, this.targetRefDB, targetRefDBIdentifier);
+						// Look for cross-references.
+						boolean xrefAlreadyExists = checkXRefExists(sourceReference, targetRefDBIdentifier);
+						if (!xrefAlreadyExists) {
+							this.logger.trace("\tCross-reference {} does not yet exist, need to create a new identifier!", targetRefDBIdentifier);
+							sourceIdentifiersWithNewIdentifier.incrementAndGet();
+							thingsToCreate.add(targetRefDBIdentifier + "," + String.valueOf(sourceReference.getDBID()) + "," + speciesID);
+						} else {
+							sourceIdentifiersWithExistingIdentifier.incrementAndGet();
```

Comment:
This line exists in both the if and the else branch, so it can be moved outside of the condition blocks (and the else block can be removed).

Reply:
They are actually different objects -- one is`sourceIdentifiersWith'New'Identifier` and the other is `sourceIdentifiersWith'Existing'Identifier`

Reply:
Yeah, you’re right.  I misread it 🙂!